ocelot/optics/wave.py

```diff
@@ -2730,6 +2730,109 @@ def dfl_prop(dfl, z, fine=1, debug=1):
 
     return dfl_out
 
+def dfl_prop_iris(dfl, N=10, a=0.055, center=(0,0), b=0.3, n_iter=100, 
+              absorption_outer_pipe=False, acount_first_cell_loss=False):
+    '''
+    Propagates radiation through an iris line.
+    
+    Parameters
+    ----------
+    dfl : RadiationField
+        Input RadiationField object.
+    
+    N : int, optional
+        Number of irises. Default is 10.
+    
+    a : float, optional
+        Iris radius. Default is 0.055.
+    
+    center : array-like of floats, optional
+        Center of each individual iris. Default for a single iris is (0,0), but for a set of irises, it should be (np.array(N), np.array(N)).
+    
+    b : float, optional
+        Spacing between irises. Default is 0.3.
+    
+    n_iter : int, optional
+        Number of propagation iterations. Must be divisible by N. Default is 100.
+    
+    absorption_outer_pipe : bool, optional
+        If the outer pipe absorbs radiation. Note: This may not represent accurate boundary conditions. Default is False.
+    
+    account_first_cell_loss : bool, optional
+        Whether to account for absorption on the first screen when calculating total loss. Default is False.
+    
+    Returns
+    -------
+    dfl_waveguide : RadiationField
+        Output RadiationField object.
+    
+    E_x_array : array-like of floats
+        x-component of the output field.
```

Comment:
One may think this is the horizontal polarization component

line-out of the radiation distribution along the iris line in the (x,z) plane, y=0